### PR TITLE
Support for resolving conflicts when pulling from remote git branch. [INS-4550]

### PR DESCRIPTION
**Pull request by yaoweiprc**
**Changes:**

- Users can now resolve conflicts in app when `pulling from remote branch`.
- You must commit your local changes first to avoid losing data.
- For each conflict file, you can choose either your version or their version.
- A new commit will be created after resolving conflicts.
image

### CODE REVIEW

**Comment by filfreire** [comment]
rebased with latest develop and fixed lint issue

**Comment by CurryYangxx** [comment]
Could you add some description to let us know what changes were made?

**Review by gatzjames — DISMISSED** [review]

**Comment by gatzjames** [comment]
Did some testing and it works great! LGTM!

Next steps for Git:
- [ ] Use the same mechanism for merge conflicts when merging branches
- [ ] Show alert when users switch branches and have changes that will be lost
- [ ] Expand e2e tests to cover these cases

File: packages/insomnia/src/sync/git/git-vcs.ts [hunk 1]
```diff
@@ -624,12 +625,171 @@ export class GitVCS {
...
+      throw new Error('Cannot pull with uncommitted changes, please commit local changes first.');
+    }
     console.log('[git] Pull remote=origin', await this.getCurrentBranch());
     return git.pull({
       ...this._baseOpts,
       ...gitCallbacks(gitCredentials),
       remote: 'origin',
       singleBranch: true,
+    }).catch(
+      async err => {
```
Comment (CurryYangxx) [inline packages/insomnia/src/sync/git/git-vcs.ts:640]: I suggest that the logic in catch can be split into several separate functions. The current function is too long and difficult to read.

File: packages/insomnia/src/sync/git/git-vcs.ts [hunk 2]
```diff
@@ -624,12 +625,171 @@ export class GitVCS {
...
+    handledMergeConflicts,
+    commitMessage,
+    commitParent,
+  }: {
+    gitCredentials?: GitCredentials | null;
+    handledMergeConflicts: MergeConflict[];
+    commitMessage: string;
+    commitParent: string[];
+  }) {
+    console.log('[git] continue to merge after resolving merge conflicts', await this.getCurrentBranch());
```
Comment (CurryYangxx) [inline packages/insomnia/src/sync/git/git-vcs.ts:788]: Does this log need to be deleted?

Comment (yaoweiprc) [reply packages/insomnia/src/sync/git/git-vcs.ts:788]: We do need this log.

File: packages/insomnia/src/sync/git/git-vcs.ts (hunk 1 above)
Comment (yaoweiprc) [reply packages/insomnia/src/sync/git/git-vcs.ts:640]: https://github.com/Kong/insomnia/pull/8118/commits/ff49aacd4e8cc9118a4c4718c8fb3d38ab3ee9d6

**Review by CurryYangxx — APPROVED** [review]

**Pull request merged**